### PR TITLE
Fix scrollbar contrast when in dark mode, by relying on built in color-scheme

### PR DESCRIPTION
**Pull request by NullVoxPopuli**
Docs: https://developer.mozilla.org/en-US/docs/Web/CSS/color-scheme

Also included:
- `npx ember-apply tailwind`, which gives the test app Tailwind 3 + JIT in the least invasive way, in that it uses the Tailwind CLI without integrating with webpack or anything as [this approach does](https://discuss.emberjs.com/t/ember-modern-css/19614) (which is really good! just not what we want since we need to support pre-embroider workflows as well). The main advantage to this native tailwind CLI approach is that it works well with Ember-CLI's CSS-hot-reloading (both with and without embroider), so we can tweak the config and CSS without re-loading the page we're viewing.
- a demo page for the manual test, and potentially something for us to run Percy against later

## How this works:

_Mac OS Settings_

light
 - always: works
 - when scrolling: we can't control this

dark
 - always: works 
 - when scrolling: we can't control this

firefox:
 - correct colors, no hover styles

----------------------------------------------------------

Things for later, 
 - (and @simonihmig and I have started talking about this), but we need to set up a central renovate config for all our open source repos to extend, as we have enough now, where some things have begun to slip through the cracks as we iterate and forget to propagate changes. In this case, regular test-app dependency updates had been disabled because at one point we thought it was a good idea to keep the tests using the oldest dependencies we wanted to support. We haven't done that in a long time though, and if there is anything we actually _need_ to support, we should declare it so, explicitly in the ember-try.js config.
 - we may want to create multiple test / example apps for our presently supported tailwind integration scenarios, as right now they are only documented on the README -- it would be good to get thees approaches integrated in to C.I.
   - Tailwind 2 + ember-cli-postcss -- non-embroider
   - Tailwind 2 + ember-css-modules -- non-embroider
   - Tailwind 3 + JIT + tailwind-CLI -- with and without embroider
   - Tailwind 3 + JIT + [webpack CSS](https://discuss.emberjs.com/t/ember-modern-css/19614) -- embroider-only
   (note that Tailwind 3 + JIT is not possible to support with ember-cli-postcss)
- it may make sense to pull this package (and its test app) in to the toucan-core monorepo at some point -- especially if we can work through the `dependenciesMeta.*.injected` issue [described here](https://github.com/CrowdStrike/ember-headless-table/pull/44)   

-----------------------------------------------------------
   
**How to test manually**
(This project does not have a docs-site atm)

- checkout this branch
- `pnpm i`
- in 3 terminals:
  - `pnpm build`
  - `cd test-app && pnpm tailwind:watch`
  - `cd test-app && pnpm start`

### CODE REVIEW

File: ember-toucan-styles/src/services/theme-manager.ts [hunk 1]
```diff
@@ -146,6 +146,16 @@ export default class ThemeManagerService extends Service {
   selectTheme(theme: Theme, { shouldSaveTheme = true } = {}) {
     document.body.classList.add(theme);
 
+    /**
+     * Sync the color scheme:
+     *   https://developer.mozilla.org/en-US/docs/Web/CSS/color-scheme
+     */
+    if (theme === DARK) {
```
Comment (NullVoxPopuli) [inline ember-toucan-styles/src/services/theme-manager.ts:153]: this is the actual fix

**Comment by simonihmig** [comment]
LGTM!

TS tests fail with `Property 'instance' must be of type 'EmberObject', but here has type 'EmberObject'.` 😅
Possibly duplicated `@types` packages with different versions?

**Comment by NullVoxPopuli** [comment]
> Possibly duplicated @types packages with different versions?

likely, been having a big problem with that with all the `@types/` related stuff lately

**Comment by NullVoxPopuli** [comment]
atm, I'm getting this error:
```
❯ pnpm tsc
error TS2688: Cannot find type definition file for 'ember__test-helpers'.
  The file is in the program because:
    Entry point for implicit type library 'ember__test-helpers'
```
which, I've gotten on other projects, but I forgot the resolution.

**Comment by NullVoxPopuli** [comment]
This has gotten crazy enough, where I'm going to pull out all the basic maintenance stuff out to a separate PR.

I think there is some incompatibility with various `@types/ember*` packages and the ember* packages that now provide their own types.. :-\

**Comment by simonihmig** [comment]
> which, I've gotten on other projects, but I forgot the resolution.

Wasn't it updating to latest `@ember/test-helpers` (which ships its own types) and removing `@types/ember__test-helpers`?

**Comment by NullVoxPopuli** [comment]
> Wasn't it updating to latest

yeah, there were a couple other dependencies that ended up needing to be updated as well as they were also bringing in the old `@types/ember__test-helpers`

**Review by ynotdraw — APPROVED** [review]
Scrollbars looked good to me in the `test-app`.

(Posting a video here of the main content section, but also looked in the smaller scrollable section as well)
https://user-images.githubusercontent.com/8069555/217679582-5e3bd9f9-4d74-4c72-a106-add9f5490a6f.mov

**Comment by ynotdraw** [comment]
> a demo page for the manual test, and potentially something for us to run Percy against later

I love this idea!

> we may want to create multiple test / example apps for our presently supported tailwind integration scenarios, as right now they are only documented on the README -- it would be good to get thees approaches integrated in to C.I.

For sure, I can definitely see value in this.

> it may make sense to pull this package (and its test app) in to the toucan-core monorepo at some point -- especially if we can work through the dependenciesMeta.*.injected issue https://github.com/CrowdStrike/ember-headless-table/pull/44

Saad had a question about this today as well, so nice you're touching on it here!  I agree that it'd be nice to move it there eventually! 🙇 To your point above about Percy, then we can _really_ exercise Toucan as a whole (both styling + components) - I dig it!

File: test-app/package.json [hunk 2]
```diff
@@ -11,20 +11,23 @@
...
-    "test:ember": "ember test"
+    "test:ember": "ember test",
+    "tailwind:watch": "npx tailwindcss -c ./config/tailwind/tailwind.config.js -i ./config/tailwind/tailwind-input.css -o ./public/assets/tailwind.css --watch",
+    "tailwind:build": "npx tailwindcss -c ./config/tailwind/tailwind.config.js -i ./config/tailwind/tailwind-input.css -o ./public/assets/tailwind.css",
+    "lint:prettier:fix": "prettier -w .",
+    "lint:prettier": "prettier -c ."
   },
   "dependencies": {
+    "@crowdstrike/tailwind-toucan-base": "^3.3.1",
     "@crowdstrike/ember-toucan-styles": "workspace:../ember-toucan-styles",
```
Comment (simonihmig) [inline test-app/package.json:31]: FWIW, I have seen this mostly as `workspace:*`, which would be less brittle towards file layout changes I guess?

**Review by simonihmig — APPROVED** [review]

File: test-app/package.json (hunk 2 above)
Comment (NullVoxPopuli) [reply test-app/package.json:31]: yeah, I have no preference here 🤷

**Review by nicolechung — APPROVED** [review]

**Pull request merged**